src/main/java/org/broadinstitute/hellbender/tools/copynumber/CondenseReadCounts.java

```diff
@@ -0,0 +1,203 @@
+            minValue = 0,
+            optional = true
+    )
+    private int expectedBinLength = 0;
+
+    @Override
+    public String doWork() {
+
+        try (final SimpleCountReader reader = new SimpleCountReader(input);
+             final SimpleCountWriter writer = new SimpleCountWriter(output)) {
+            String contig = null;
+            int firstPosition = -1;
+            int lastPosition = -1;
+            int currentReadCount = 0;
+            int currentRecordCount = 0;
+            SimpleCount record;
+            int skippedRecords = 0;
+            double skippedReadCounts = 0;
+            double outputCounts = 0;
+            int condensedRecords = 0;
+            while ((record = reader.readRecord()) != null) {
+                if (record.getContig().equals(contig) && record.getStart() == lastPosition + 1) {
+                    currentReadCount += record.getCount();
+                    lastPosition = record.getEnd();
+                    if (++currentRecordCount == factor) {
+                        if (expectedBinLength > 0 && (lastPosition - firstPosition + 1) == expectedBinLength) {
+                            if (condensedRecords == 0) {
+                                writer.writeHeaderComments(reader.headerLines());
+                            }
+                            writer.writeRecord(composeRecord(contig, firstPosition, lastPosition, currentReadCount));
+                            condensedRecords += currentRecordCount;
+                            outputCounts += currentReadCount;
+                        } else {
+                            skippedReadCounts += currentReadCount;
+                            skippedRecords += currentRecordCount;
+                        }
+                        contig = null;
+                        currentRecordCount = 0;
+                        currentReadCount = 0;
+                        firstPosition = lastPosition = -1;
+                    }
+                } else {
+                    skippedRecords += currentRecordCount;
+                    skippedReadCounts += currentReadCount;
+                    contig = record.getContig();
+                    firstPosition = record.getStart();
+                    lastPosition = record.getEnd();
+                    currentReadCount = record.getCount();
+                    currentRecordCount = 1;
+                }
+
+            }
+            skippedRecords += currentRecordCount;
+            skippedReadCounts += currentReadCount;
+            logger.info("Number of output records: " + condensedRecords / factor);
+            logger.info("Number of condensed input records: " + condensedRecords);
+            logger.info("Number of skipped input records: " + skippedRecords);
+            logger.info("Total count condensed: " + outputCounts);
+            logger.info("Total count skipped: " + skippedReadCounts);
+            logger.info(String.format("%% of lossed records and counts are: %.2g and %.2g",
```

Comment:
Why "lossed" here and "skipped" above?

Reply:
I would also just break this message up into two lines to be consistent with the above.

Reply:
lossed -> lost